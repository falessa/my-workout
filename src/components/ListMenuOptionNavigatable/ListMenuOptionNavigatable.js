import React from 'react';
import { StyleSheet, View, Text, FlatList } from 'react-native';
import { useTranslation } from 'react-i18next';
import { TouchableOpacity } from 'react-native-gesture-handler';
import Settings from '../../utils/SettingsEnum'

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff',
        alignItems: 'stretch',
        justifyContent: 'center',
        width: '95%',
        borderBottomWidth: 1,
        borderBottomColor: '#eee',
        padding: 10
    },
    text: {
        fontSize: 18,
    }
})

const ListMenuOption = ({ options, navigation }) => {
    const { t } = useTranslation()

    const navigateToMenuScreen = item => {
        switch(item) {
            case Settings.LANGUAGE: {
                navigation.navigate('LanguageScreen')
                break
            }
            case Settings.HELP: {
                navigation.navigate('HelpScreen')
                break
            }
            case Settings.ABOUT_US: {
                console.log('ABOUT US SCREEN NOT IMPLEMENTED YET')
                break
            }
            default: {
                throw new Error('Setting does not exist')
            }
        }
    }

    return (
        <View style={styles.container}>
            <FlatList
                data={options}
                keyExtractor={x => String(x)}
                renderItem={({ item }) => 
                    <TouchableOpacity style={styles.container} onPress={() => navigateToMenuScreen(item)}>
                        <Text style={styles.text}>{t(item)}</Text>
                    </TouchableOpacity>
                }
            />
        </View>
    )
}

export default ListMenuOption;