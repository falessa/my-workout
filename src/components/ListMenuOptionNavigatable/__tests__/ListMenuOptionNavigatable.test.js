import "react-native";
import React from "react";
import { create } from "react-test-renderer";
import ListMenuOptionNavigatable from '../ListMenuOptionNavigatable';

describe.skip('ListMenuOptionNavigatable tests', () => {
    const props = {
        options: [
            "Option 1",
            "Option 2",
            "Option 3"
        ],
        navigation: () => {
            // nothing
        }
    }

    const wrapper = create(<ListMenuOptionNavigatable {...props} />)

    test('Render ListMenuOptionNavigatable', () => {
        expect(wrapper).toMatchSnapshot()
    })
})