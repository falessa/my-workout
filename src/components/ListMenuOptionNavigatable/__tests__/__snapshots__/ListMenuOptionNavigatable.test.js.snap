// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListMenuOptionNavigatable tests Render ListMenuOptionNavigatable 1`] = `
<View
  style={
    {
      "alignItems": "stretch",
      "backgroundColor": "#fff",
      "borderBottomColor": "#eee",
      "borderBottomWidth": 1,
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
      "width": "95%",
    }
  }
>
  <RCTScrollView
    data={
      [
        "Option 1",
        "Option 2",
        "Option 3",
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <RNGestureHandlerButton
          collapsable={false}
          delayLongPress={600}
          enabled={true}
          exclusive={true}
          handlerTag={1}
          handlerType="NativeViewGestureHandler"
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
          touchSoundDisabled={false}
        >
          <View
            accessible={true}
            collapsable={false}
            style={
              {
                "alignItems": "stretch",
                "backgroundColor": "#fff",
                "borderBottomColor": "#eee",
                "borderBottomWidth": 1,
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 10,
                "width": "95%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 18,
                }
              }
            >
              Option 1
            </Text>
          </View>
        </RNGestureHandlerButton>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <RNGestureHandlerButton
          collapsable={false}
          delayLongPress={600}
          enabled={true}
          exclusive={true}
          handlerTag={2}
          handlerType="NativeViewGestureHandler"
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
          touchSoundDisabled={false}
        >
          <View
            accessible={true}
            collapsable={false}
            style={
              {
                "alignItems": "stretch",
                "backgroundColor": "#fff",
                "borderBottomColor": "#eee",
                "borderBottomWidth": 1,
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 10,
                "width": "95%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 18,
                }
              }
            >
              Option 2
            </Text>
          </View>
        </RNGestureHandlerButton>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <RNGestureHandlerButton
          collapsable={false}
          delayLongPress={600}
          enabled={true}
          exclusive={true}
          handlerTag={3}
          handlerType="NativeViewGestureHandler"
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
          touchSoundDisabled={false}
        >
          <View
            accessible={true}
            collapsable={false}
            style={
              {
                "alignItems": "stretch",
                "backgroundColor": "#fff",
                "borderBottomColor": "#eee",
                "borderBottomWidth": 1,
                "flex": 1,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 10,
                "width": "95%",
              }
            }
          >
            <Text
              style={
                {
                  "fontSize": 18,
                }
              }
            >
              Option 3
            </Text>
          </View>
        </RNGestureHandlerButton>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
